addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -143,12 +154,219 @@ odoo.define('website_form.s_website_form', function (require) {
             // Reinitialize dates
             this.$allDates.removeClass('s_website_form_datepicker_initialized');
         },
+        /**
+        * calculates the visibility at each event input on the form
+        * @param {Object} event
+        */
+        onFormInputChange: function (event) {
+            document.querySelectorAll('[data-visibility-dependency]').forEach((input)=>{
+                const idOfDependency = input.getAttribute('data-visibility-dependency');
+                if (idOfDependency) {
+                    const dependecyField = document.getElementById(idOfDependency);
+                    if (dependecyField) {
+                        this._manageVisibility(dependecyField, input);
+                    }
+                }
+            });
+        },
+        /**
+        * checks the data visibility condition and show/hide field
+        *
+        * @param {HTMLElement} dependency the field that is concerned by the visibility condition
+        * @param {HTMLElement} input the input to hide or to show depends of visibility condition
+        */
+        _manageVisibility: function (dependency, input) {
+            if (!input) {
+                return;
+            }
+            let field = dependency;
+            while (!field.classList.contains('s_website_form_field')) {
+                field = field.parentElement;
+            }
+            if (field.classList.contains('s_website_form_field_hidden_if') && field.classList.contains('hidden')) {
+                //The dependecy is hidden so this field is hidden too
+                return this._hideField(input);
+            }
+            const visibilityCondition = input.getAttribute('data-visibility-condition');
+            const comparator = input.getAttribute('data-visibility-comparaison');
+            if (['radio', 'checkbox'].includes(dependency.type)) {
+                dependency = document.getElementById(visibilityCondition);
+            }
+            const currentValue = this._readValueOfInput(dependency);
+            if (dependency.type === 'checkbox' || dependency.type === 'radio') {
+                if (this._compareTo(currentValue, comparator)) {
+                    return this._showField(input);
+                }
+                return this._hideField(input);
+            }
+            if (dependency.nodeName === 'SELECT') {
+                const keyOption = dependency.querySelectorAll('option')[visibilityCondition];
+                if (this._compareTo(currentValue, `select${comparator}`, keyOption.value)) {
```

Comment:
I don't understand this ``` `select${comparator}` ``` comparator? :thinking: Anyways, we come back to the same idea explained in my above comment :point_up: This conditions should be written "is dependency select field's value equal to 'abcd' ?" not "is the option with value 'abcd' in my dependency select field selected?", I think this can simplify the code... but tell me if I am wrong, I can be.

Reply:
Since visibility dependencies are managed via ids, I can't think of a better way to handle this. But maybe I'm missing something...

I use `select${comparator} `because with the same we-select, I manage radios, checkboxes and selects comparators. And as you mentioned, getting the selected value from a select is not done in the same way.
Indeed this solution may not be the most beautiful but seems to me to be a lesser evil compared to duplicate all the comparator of the we-select. Maybe there is a way to manage this in a cleaner way but I have no idea for the moment :/

Reply:
I am not sure to understand everything but I think this is the same idea of using value by name using the JavaScript API (see my other comment). Maybe the dependencies should just totally be written in terms of field names instead of field IDs actually.